olad/OlaServer.cpp

```diff
@@ -472,9 +476,17 @@ bool OlaServer::InternalNewConnection(
 }
 
 void OlaServer::ReloadPluginsInternal() {
+#ifdef HAVE_LIBSYSTEMD
+  // Return value is intentionally not checked for both calls.
+  // See return value section under sd_notify(3).
+  sd_notify(0, "RELOADING=1\nSTATUS=Reloading plugins\n");
```

Comment:
Having read the man page, I understand why it's not checked, but for user sanity, we should add some logging. Probably both to signify sd_notify has been run/the return code and particularly a warning if it failed.

I'd probably suggest OLA_WARN if we get <0 return code, and then maybe OLA_INFO for 0 and OLA_DEBUG for >0, but that may be a bit excessive. Likewise others may argue those log levels should all be reduced by one...

Reply:
What happens if the system is too old to support RELOADING? I assume they just get discarded as they're plain text?

Reply:
>What happens if the system is too old to support RELOADING? I assume they just get discarded 
>as they're plain text?

`sd_notify(3)` mentions this: `The string may contain any kind of variable assignments, but the following shall be considered well-known`.

I think that does indeed imply that if the service manager doesn't recognize the variable being assigned to, it'll either log that somewhere or ignore that assignment, so we shouldn't have a problem with old systems.

>  I understand why it's not checked, but for user sanity, we should add some logging. Probably 
>  both to signify sd_notify has been run/the return code and particularly a warning if it failed.

If that's the case, would a message at startup do? `libsystemd` provides capabilities for determining whether a binary has been setup under the control of `systemd`, and that could be used to indicate that notification messages are being used. As for the failures, though, we could log those as well, but I think conditionally - only when being run under `systemd`

Reply:
Scratch that, there's no library function to determine whether the service has been run under systemd, but the call shouldn't return a negative value if the socket was not passed, anyway. I'll log failures and log from main to indicate whether notifications are being passed.